[nnc][quantization] Optimize constructTensors in ext functions

IvanKobzarev opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70081
* #70080
* #69954
* __->__ #69856

Differential Revision: [D33064756](https://our.internmc.facebook.com/intern/diff/D33064756)

IvanKobzarev committed:
[nnc][quantization] Optimize constructTensors in ext functions

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

IvanKobzarev committed:
Update on "[nnc][quantization] Optimize constructTensors in ext functions"


Differential Revision: [D33064756](https://our.internmc.facebook.com/intern/diff/D33064756)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

IvanKobzarev committed:
Update on "[nnc][quantization] Optimize constructTensors in ext functions"


Differential Revision: [D33064756](https://our.internmc.facebook.com/intern/diff/D33064756)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

kimishpatel commented on torch/csrc/jit/tensorexpr/external_functions.cpp:
```diff
@@ -119,6 +72,110 @@ at::Tensor from_blob_quantized(
+                             // NOLINTNEXTLINE
+                             buf_dims_vec[i]))
+                         .requires_grad(false);
+      if (auto qd = qdata[i]) {
```
Can you separate out assign from the if for better readability?

IvanKobzarev replied:
@kimishpatel Why does it make code less readable ? :) 
I thought the main point of introducing initialization inside the if conditions was to increase readability:
https://skebanga.github.io/if-with-initializer/

kimishpatel replied:
I was mainly coming from "hey is this a typo?" and then I realized no it isnt. The argument in the link seems to be more from the perspective of more verbose code specifically due to the usage pattern it has. I am fine if you want to keep it this way. It is matter of personal preference.

ZolotukhinM reviewed (approved):
Overall looks good to me, but I'm not sure about the sigmoid change. Is there a way we can avoid it?

IvanKobzarev committed:
Update on "[nnc][quantization] Optimize constructTensors in ext functions"


Differential Revision: [D33064756](https://our.internmc.facebook.com/intern/diff/D33064756)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[nnc][quantization] Optimize constructTensors in ext functions"


Differential Revision: [D33064756](https://our.internmc.facebook.com/intern/diff/D33064756)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

IvanKobzarev committed:
Update on "[nnc][quantization] Optimize constructTensors in ext functions"


Differential Revision: [D33064756](https://our.internmc.facebook.com/intern/diff/D33064756)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

IvanKobzarev committed:
Update on "[nnc][quantization] Optimize constructTensors in ext functions"


Differential Revision: [D33064756](https://our.internmc.facebook.com/intern/diff/D33064756)

[ghstack-poisoned]

IvanKobzarev commented:
@ivankobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).

IvanKobzarev committed:
Update on "[nnc][quantization] Optimize constructTensors in ext functions"


Differential Revision: [D33064756](https://our.internmc.facebook.com/intern/diff/D33064756)

[ghstack-poisoned]

IvanKobzarev committed:
Update on "[nnc][quantization] Optimize constructTensors in ext functions"


Differential Revision: [D33064756](https://our.internmc.facebook.com/intern/diff/D33064756)

[ghstack-poisoned]

IvanKobzarev commented:
@IvanKobzarev has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33064756).
